fulp_modules/features/clothing/ballgown/autodrobe.dm

```diff
@@ -0,0 +1,251 @@
+/obj/machinery/vending/autodrobe
```

Comment:
Instead of completely redefining the autodrobe, you should use the `Initialize` proc to add onto the list of products, then call the original `Initialize` proc. 

Something like this (i wrote this on my phone and its probably incomplete, don't just copy/paste):
```dreammaker
/obj/machinery/vending/autodrobe/Initialize()
    product_categories[4] += [/obj/item/clothing/under/dress/ballgown = 2]
    ..() // <- Calls the original definition of the proc
```